src/classes/frCampaign.cls

```diff
@@ -76,27 +76,54 @@ public class frCampaign {
             '. Exception: ' + e.getMessage());
         }
     }
-
+    
     public void create(Map<String, Object> request) {
         String goalId = String.valueOf(request.get('goalId'));
-        try {
-            Campaign newCampaign = new Campaign();
-            newCampaign.Name = String.valueOf(request.get('name'));
-            newCampaign.Description = String.valueOf(request.get('reason'));
-            newCampaign.Status = String.valueOf(request.get('status'));
-            newCampaign.ExpectedRevenue = Decimal.valueOf(String.valueOf(request.get('goal')));
-            newCampaign.fr_ID__c = goalId;
-
-            String parentId = String.valueOf(request.get('parentGoalId'));
-            if (parentId != null && parentId != '0') {
-                Campaign parentCampaign = Database.query('select Id, fr_ID__c from Campaign where fr_ID__c = :parentId');
-                newCampaign.ParentId = parentCampaign.Id;
+       
+        Campaign newCampaign = new Campaign();
+        newCampaign.Name = String.valueOf(request.get('name'));
+        newCampaign.Description = String.valueOf(request.get('reason'));
+        newCampaign.Status = String.valueOf(request.get('status'));
+        newCampaign.ExpectedRevenue = Decimal.valueOf(String.valueOf(request.get('goalAmount')));
+        newCampaign.fr_ID__c = goalId;
+        
+        String parentId = String.valueOf(request.get('parentGoalId'));
+        if (parentId != null && parentId != '0') {
+            Campaign parentCampaign = Database.query('select Id, fr_ID__c from Campaign where fr_ID__c = :parentId');
+            if(parentCampaign != null) {
+                newCampaign.ParentId = parentCampaign.Id;                
             }
+        }
+         try {
             Database.upsert(newCampaign, Campaign.Field.fr_ID__c, true);
         } catch (DMLException e) {
         	insert new Error__c(Error__c =
             'Failed to insert funraise campaign for campaign id ' + goalId +
             '. Exception: ' + e.getMessage());
+            return;
+        }
+        
+        String supporterId = String.valueOf(request.get('supporterId'));
+        if(String.isNotBlank(supporterId)) {
```

Comment:
Should this use the same type of check as line 91?

```suggestion
        if(supporterId != null && supporterId != '0') {
```

In platform our SFDCCampaignMessage DTO has `parentGoalId` and `supporterId` both as primitive types: `long`

Reply:
https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_methods_system_string.htm

We're good on checking if it's null since isBlank/isNotBlank will check:

> Returns true if the specified String is white space, empty (''), or null; otherwise, returns false.

don't think we can use this check on line 91 though because there's some logic there around it not being the string "0" which I'd like to reinvestigate why that's happening but in a different PR

Reply:
Oh I checked the platform logic and you're right, in ToSFDCCampaignMessage we'll send 0 for the supporterId here if it isn't a fundraising page. I'll add that check then